parsers/NORDPOOL.py

```diff
@@ -0,0 +1,196 @@
+from datetime import datetime, timedelta
+from enum import Enum
+from functools import lru_cache
+from logging import Logger, getLogger
+
+from requests import Response, Session
+
+from electricitymap.contrib.lib.models.event_lists import PriceList
+from electricitymap.contrib.lib.types import ZoneKey
+
+from .lib.utils import get_token
+
+NORDPOOL_BASE_URL = "https://data-api.nordpoolgroup.com/api/v2/"
+
+# Set this as a global const outside of the functions to cache the token for the current parser instance
+NORDPOOL_TOKEN = None
+NORDPOOL_TOKEN_EXPIRATION = None
+
+SOURCE = "nordpool.com"
+
+
+class NORDPOOL_API_ENDPOINT(Enum):
+    PRICE = "Auction/Prices/ByAreas"
+
+
+class MARKET_TYPE(Enum):
+    DAY_AHEAD = "DayAhead"
+    GB_DAY_AHEAD = "GbHalfHour_DayAhead"
+
+
+class CURRENCY(Enum):
+    EUR = "EUR"
+    DKK = "DKK"
+    SEK = "SEK"
+    NOK = "NOK"
+    GBP = "GBP"
+
+
+ZONE_MAPPING = {
+    "AT": "AT",
+    "BE": "BE",
+    "DK-DK1": "DK1",
+    "DK-DK2": "DK2",
+    "DE": "GER",
+    "EE": "EE",
+    "FI": "FI",
+    "FR": "FR",
+    "GB": "UK",
+    "LT": "LT",
+    "LV": "LV",
+    "NL": "NL",
+    "NO-NO1": "NO1",
+    "NO-NO2": "NO2",
+    "NO-NO3": "NO3",
+    "NO-NO4": "NO4",
+    "NO-NO5": "NO5",
+    "PL": "PL",
+    "SE-SE1": "SE1",
+    "SE-SE2": "SE2",
+    "SE-SE3": "SE3",
+    "SE-SE4": "SE4",
+}
+
+INVERTED_ZONE_MAPPING = {value: key for key, value in ZONE_MAPPING.items()}
+
+
+def _generate_new_nordpool_token(session: Session | None = None) -> str:
+    global NORDPOOL_TOKEN_EXPIRATION
```

Comment:
In general, global variables are not the best because a variable might be modified unexpectedly in another context. 
If we keep it only to this file and it doesn't grow too much this should be ok. However, if we start fetching a lot of different types in parallel using code from this file this might cause problems in the long run. So overall, I'm ok for now but we should be open to other solutions.